src/Json/Value.php

```diff
@@ -635,37 +636,80 @@ public function ¿timestamp(): ?DateTimeImmutable
     }
 
     /**
-     * Assert this value is a string, assert it is any of the names of the given Enum, then return that Enum instance.
+     * Assert this value is a valid value for the provided enum class, returning the corresponding enum case. Supports
+     *  string- and int-backed enums. Also supports enums from the package `vcn/enum`.
      *
-     * @template T of Enum
+     *  For string- and int-backed enums, the JSON value should equal the type and value of a value backing an enum case.
+     *
+     *  For enums from package `vcn/enum`, the value should be a string and should equal the name of the enum case.
+     *
+     * @template T of BackedEnum|Enum
```

Comment:
This documentation might be clearer by reusing the existing "backing" vocabulary, superimposing it over `vcn/enum` and by pointing out our chosen semantics:

```
    /**
     * Assert this value backs a given enum, then return that enum case.
     *
     * Supports both native backed enums and those defined using the `vcn/enum` package.
     *
     * Does **not** support native pure enums.
     *
     * If given a native backed enum, case construction uses `BackedEnum::from`, **requiring this value to be strict
     * equal to the declared backing value**.
     *
     * If given an enum defined in the `vcn/enum` package, case (instance) construction uses `Vcn\Lib\Enum::byName`.
     *
     * ...
     *
     * @throws Exception\AssertionFailed If this value does not back $className.
     * @throws Exception\Runtime         If $className does not exist, does not extend BackedEnum, or does not extend
     *                                   Vcn\Lib\Enum.
     *
     * @see https://www.php.net/manual/en/language.enumerations.backed.php
     */
```

Reply:
is indeed much clearer; fixed